buildSrc/src/main/java/com/google/firebase/gradle/plugins/Metalava.kt

```diff
@@ -14,83 +14,203 @@
 
 package com.google.firebase.gradle.plugins
 
-import com.android.build.gradle.api.AndroidSourceSet
 import java.io.File
+import java.io.FileOutputStream
+import java.io.OutputStream
 import org.gradle.api.DefaultTask
 import org.gradle.api.Project
 import org.gradle.api.artifacts.Configuration
 import org.gradle.api.file.FileCollection
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.SetProperty
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
 import org.gradle.api.tasks.InputFiles
-import org.gradle.api.tasks.Nested
 import org.gradle.api.tasks.OutputDirectory
-import org.gradle.api.tasks.SourceSet
+import org.gradle.api.tasks.OutputFile
 import org.gradle.api.tasks.TaskAction
 
 val Project.metalavaConfig: Configuration
     get() =
         configurations.findByName("metalavaArtifacts")
-                ?: configurations.create("metalavaArtifacts") {
-                    this.dependencies.add(this@metalavaConfig.dependencies.create("com.android:metalava:1.3.0"))
-                }
+            ?: configurations.create("metalavaArtifacts") {
+                this.dependencies.add(this@metalavaConfig.dependencies.create("com.android.tools.metalava:metalava:1.0.0-alpha06"))
+            }
 
 fun Project.runMetalavaWithArgs(
-    arguments: List<String>
+    arguments: List<String>,
+    ignoreFailure: Boolean = false,
+    stdOut: OutputStream? = null
 ) {
     val allArgs = listOf(
-            "--no-banner",
-            "--hide",
-            "HiddenSuperclass" // We allow having a hidden parent class
+        "--no-banner",
+        "--hide",
+        "HiddenSuperclass", // We allow having a hidden parent class
+        "--hide",
+        "HiddenAbstractMethod"
```

Comment:
why was this add?

Reply:
This is a new check metalava added, similar to the one you fixed in piper for auth